File: tests/typesafety/python/email_checked.py
# expect: passes
# title: Both senders check the address again
def looks_like_email(raw: str) -> bool:
    return "@" in raw and not raw.startswith("@")


def send_receipt(to: str) -> str:
    if not looks_like_email(to):
        raise ValueError("bad address")
    return f"receipt sent to {to}"


def send_reminder(to: str) -> str:
    if not looks_like_email(to):
        raise ValueError("bad address")
    return f"reminder sent to {to}"

File: tests/typesafety/typescript/email_checked.ts
// expect: passes

function looksLikeEmail(raw: string): boolean {
  return raw.includes("@") && !raw.startsWith("@");
}

export function sendReceipt(to: string): string {
  if (!looksLikeEmail(to)) {
    throw new Error("bad address");
  }
  return `receipt sent to ${to}`;
}

export function sendReminder(to: string): string {
  if (!looksLikeEmail(to)) { // the same check, again
    throw new Error("bad address");
  }
  return `reminder sent to ${to}`;
}
